test/test_interface_elements.jl

```diff
@@ -0,0 +1,390 @@
+@testset "Interface elements" begin
+
+@testset "InterfaceCell" begin
+    for (here, there, sdim, rdim, shape, rshape) in (
+            (         Line((1,2)),            Line((3,4)),   2, 1, RefQuadrilateral, RefLine), 
+            (QuadraticLine((1,2,5)),          Line((3,4)),   2, 1, RefQuadrilateral, RefLine), 
+            (         Line((1,2)),   QuadraticLine((3,4,5)), 2, 1, RefQuadrilateral, RefLine), 
+            (QuadraticLine((1,2,5)), QuadraticLine((3,4,6)), 2, 1, RefQuadrilateral, RefLine),
+            (         Triangle((1,2,3)),                Triangle((4,5,6)),          3, 2, RefPrism, RefTriangle), 
+            (QuadraticTriangle((1,2,3,7,8,9)),          Triangle((4,5,6)),          3, 2, RefPrism, RefTriangle), 
+            (         Triangle((1,2,3)),       QuadraticTriangle((4,5,6,7,8,9)),    3, 2, RefPrism, RefTriangle), 
+            (QuadraticTriangle((1,2,3,7,8,9)), QuadraticTriangle((4,5,6,10,11,12)), 3, 2, RefPrism, RefTriangle),
+            (         Quadrilateral((1,2,3,4)),                        Quadrilateral((5,6,7,8)),                3, 2, RefHexahedron, RefQuadrilateral), 
+            (QuadraticQuadrilateral((1,2,3,4,9,10,11,12,13)),          Quadrilateral((5,6,7,8)),                3, 2, RefHexahedron, RefQuadrilateral), 
+            (         Quadrilateral((1,2,3,4)),               QuadraticQuadrilateral((5,6,7,8,9,10,11,12,13)),  3, 2, RefHexahedron, RefQuadrilateral), 
+            (QuadraticQuadrilateral((1,2,3,4,9,10,11,12,17)), QuadraticQuadrilateral((5,6,7,8,13,14,15,16,18)), 3, 2, RefHexahedron, RefQuadrilateral)
+            ) # The nodes have been chosen as the numbers representing their expected ordering
+        Chere = typeof(here)
+        Cthere = typeof(there)
+
+        @test InterfaceCell{sdim, rdim, shape, rshape, Chere, Cthere}(here, there) isa InterfaceCell{sdim, rdim, shape, rshape, Chere, Cthere}
+        @test InterfaceCell(here, there) isa InterfaceCell{sdim, rdim, shape, rshape, Chere, Cthere}
+        cell = InterfaceCell(here, there)
+
+        @test Ferrite.nvertices(cell) == Ferrite.nvertices(here) + Ferrite.nvertices(there)
+        @test Ferrite.nfaces(cell) == 2
```

Comment:
Where does this assumption come from?

Reply:
The idea is that the two sides of an interface are considered the faces of the cell.
To me it makes sense to think of it this way, but theses definitions can definitely be discussed.